src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -124,6 +127,15 @@ void fillInnerParts() {
         commandBoxPlaceholder.getChildren().add(commandBox.getRoot());
     }
 
+    /**
+     * Refresh person list panel for
+     */
+    public void refreshPersonListPanel() {
```

Comment:
Just for my understanding, how come this bit is necessary? I thought the UI anyways updated on its own when changes were made?

Reply:
This is required because the archived list is a separate filteredlist from the main filteredlist. Since personlistcard is always observing the main filteredlist as done in this line. 
` personListPanel = new PersonListPanel(logic.getFilteredPersonList());`

I will need this function along with the listener to switch personlistpanel to track my archivedpersonlist when the archive-list command is called

Reply:
Good catch, I've edited